### ISSUE TITLE
Add support for overriding the screen orientation

### ISSUE THREAD

**Issue by whimboo** [w3c/webdriver-bidi#771]
Both Puppeteer and Playwright support an `isLandscape` boolean parameter for the `setViewport` command, which is currently missing in WebDriver BiDi. We should add support for this parameter with the following behavior:

- **Default Behavior**: By default, `isLandscape` is set to `false`, meaning the orientation will not change, even if the `width` is smaller than the `height` of the requested viewport.
  
- **Landscape Mode**: If `isLandscape` is set to `true`, the provided dimensions will automatically swap when the `width` is smaller than the `height` (e.g., a setting of 800x1280 will result in a viewport of 1280x800).

- **Portrait Mode**: To force portrait mode, `isLandscape` should be set to `false`, and the dimensions must be manually specified (e.g., 800x1280).

**Comment by jgraham** [comment]
I don't really understand the use case here. If you want to set a landscape display, just set a larger width than height. One can also imagine a tablet or a fold-out phone with a square display. Is there anything else that this is controlling?

**Comment by whimboo** [comment]
We would have to let Puppeteer or Playwright folks speak up here. But I assume that this is helpful for the [`knownDevices`](https://github.com/puppeteer/puppeteer/blob/280e6f82065a2cbbe696673934a9c10610a780c6/packages/puppeteer-core/src/common/Device.ts) class so that you don't have to define all the viewports twice, but just swap the `width` and `height`.

**Comment by jgraham** [comment]
Yes, feedback from client authors would be very helpful. On the face of it I don't see why setting a boolean is much easier than swapping argument values, but maybe I'm missing some extra bit of state that this value controls.

**Comment by mxschmitt** [comment]
We [use screenOrientation](https://github.com/microsoft/playwright/blob/58bcfc2a6be059afe22c46770143ef3f275a75b6/packages/playwright-core/src/server/chromium/crPage.ts#L986), [like Puppeteer](https://github.com/puppeteer/puppeteer/blob/280e6f82065a2cbbe696673934a9c10610a780c6/packages/puppeteer-core/src/cdp/EmulationManager.ts#L280) to emulate [`screen.orientation.angle`](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/angle), [`screen.orientation.type`](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation/type) and `window.orientation` - see [here](https://github.com/microsoft/playwright/blob/5b28d2a84c0fa5232ab029ff1e6c7c73c10f9e42/tests/library/browsercontext-viewport-mobile.spec.ts#L87) and [here](https://github.com/microsoft/playwright/blob/5b28d2a84c0fa5232ab029ff1e6c7c73c10f9e42/tests/library/browsercontext-viewport.spec.ts#L179) for our tests. I think thats what it is mostly for from our side.

**Comment by OrKoN** [comment]
Puppeteer uses it to emulate https://w3c.github.io/screen-orientation/#dom-screen-orientation ([tests](https://github.com/puppeteer/puppeteer/blob/47d6c4423e61b42867840e6714567b529040593a/test/src/emulation.spec.ts#L114)) and uses the same method. A workaround for a one-time emulation can be probably done using a preload script but it might be also useful to emulate the screen orientation dynamically to test transitions from one mode to another after the page is loaded.

From the WebDriver BiDi spec perspective, we should probably closer follow the corresponding Web spec and what CDP does instead of only exposing isLandscape flag. Perhaps, we should even define the WebDriver API in https://w3c.github.io/screen-orientation/#dom-screen-orientation (although I am not sure if this feature might span multiple specifications in the future).

**Comment by jgraham** [comment]
(alternatively I suppose just allowing setting the orientation type to one of the four supported values might be enough, and the angle could be calculated).

**Comment by sadym-chromium** [comment]
Done in https://github.com/w3c/webdriver-bidi/pull/925

**Comment by whimboo** [comment]
I've updated the issue summary to better reflect what was done, and to reduce confusion. 

I assume we should be done and can close this issue?

**Comment by sadym-chromium** [comment]
I believe the requested scenario is now implementable via BiDi, and we can close this issue. Feel free to reopen this or to file a new one if something is missing.
